src/chainrules.jl

```diff
@@ -193,3 +193,108 @@ ChainRulesCore.@scalar_rule(
 ChainRulesCore.@scalar_rule(expinti(x), exp(x) / x)
 ChainRulesCore.@scalar_rule(sinint(x), sinc(invπ * x))
 ChainRulesCore.@scalar_rule(cosint(x), cos(x) / x)
+
+# non-holomorphic functions
+function ChainRulesCore.frule((_, Δν, Δx), ::typeof(besselix), ν::Number, x::Number)
+    # primal
+    Ω = besselix(ν, x)
+
+    # derivative
+    ∂Ω_∂ν = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+    ∂Ω_∂ν_Δν = ∂Ω_∂ν * Δν
+    ΔΩ = if ∂Ω_∂ν_Δν isa ChainRulesCore.NotImplemented
+        ∂Ω_∂ν_Δν
+    else
+        a = (besselix(ν - 1, x) + besselix(ν + 1, x)) / 2
+        if Δx isa Real
+            muladd(muladd(-sign(real(x)), Ω, a), Δx, ∂Ω_∂ν_Δν)
+        else
+            muladd(a, Δx, muladd(-sign(real(x)) * Ω, real(Δx), ∂Ω_∂ν_Δν))
+        end
```

Comment:
This can't be tested currently and requires https://github.com/JuliaDiff/ChainRulesCore.jl/pull/477. With this PR all definitions are tested and tests pass locally.